modules/edu.gemini.seqexec.web/edu.gemini.seqexec.web.client/src/main/scala/edu/gemini/seqexec/web/client/components/QueueTable.scala

```diff
@@ -0,0 +1,533 @@
+
+  private def stateRenderer(p: Props) = linkedTextRenderer(p) { r =>
+    <.p(SeqexecStyles.queueText, statusText(r.status, r.runningStep))
+  }
+
+  private def instrumentRenderer(p: Props) = linkedTextRenderer(p) { r =>
+    <.p(SeqexecStyles.queueText, r.instrument.show)
+  }
+
+  private val daytimeCalibrationTargetName: TagMod =
+    <.span(SeqexecStyles.daytimeCal, DaytimeCalibrationTargetName)
+
+  private def targetRenderer(p: Props) = linkedTextRenderer(p) { r =>
+    val targetName = r.targetName.fold(daytimeCalibrationTargetName)(x => x: TagMod)
+    <.p(SeqexecStyles.queueText, targetName)
+  }
+
+  private def statusIconRenderer(p: Props): CellRenderer[js.Object, js.Object, QueueRow] =
+    (_, _, _, row: QueueRow, _) => {
+      val icon: TagMod =
+        row.status match {
+          case SequenceState.Completed =>
+            IconCheckmark.copyIcon(fitted = true, extraStyles = List(SeqexecStyles.selectedIcon))
+          case SequenceState.Running(_, _) =>
+            IconCircleNotched.copyIcon(fitted = true,
+                                       loading = true,
+                                       extraStyles = List(SeqexecStyles.runningIcon))
+          case SequenceState.Failed(_) =>
+            IconAttention.copyIcon(fitted = true, extraStyles = List(SeqexecStyles.selectedIcon))
+          case _ =>
+            if (row.active)
+              IconSelectedRadio.copyIcon(fitted = true,
+                                         extraStyles = List(SeqexecStyles.selectedIcon))
+            else iconEmpty
+        }
+
+      val page = SequencePage(row.instrument, row.obsId, 0)
+      linkTo(p, page)(
+        SeqexecStyles.queueIconColumn,
+        icon
+      )
+    }
+
+  private val statusHeaderRenderer: HeaderRenderer[js.Object] = (_, _, _, _, _, _) =>
+    <.div(
+      ^.title := "Control",
+      ^.width := IconColumnWidth.px
+  )
+
+  // Renderer for a resizable column
+  private def resizableHeaderRenderer(
+      rs: (String, JsNumber) => Callback): HeaderRenderer[js.Object] =
+    (_, dataKey: String, _, label: VdomNode, _, _) =>
+      ReactFragment.withKey(dataKey)(
+        <.div(
+          ^.cls := "ReactVirtualized__Table__headerTruncatedText",
+          label
+        ),
+        Draggable(
+          Draggable.props(
+            axis = Axis.X,
+            defaultClassName = "DragHandle",
+            defaultClassNameDragging = "DragHandleActive",
+            onDrag = (ev: MouseEvent, d: DraggableData) => rs(dataKey, d.deltaX),
+            position = ControlPosition(0)
+          ),
+          <.span(^.cls := "DragHandleIcon", "⋮")
+        )
+    )
+
+  def rowClassName(p: Props)(i: Int): String =
+    ((i, p.rowGetter(i)) match {
+      case (-1, _) =>
+        SeqexecStyles.headerRowStyle
+      case (_, QueueRow(_, s, _, _, _, _, _)) if s == SequenceState.Completed  =>
+        SeqexecStyles.stepRow |+| SeqexecStyles.rowPositive
```

Comment:
I assume this means `SeqexecStyles` has a monoid type class? Nice.

Reply:
`GStyle`